src/controllers/users.controller.ts

```diff
@@ -35,11 +33,21 @@ export class UserController {
 
   @Post()
   async post(@Body() user: User): Promise<InsertResult> {
-    return this.usersService.createUser(user);
+    try {
+      const createdUser = await this.usersService.createUser(user);
+      return createdUser;
+    } catch (error) {
+      throw new BadRequestError(
+        error?.detail ?? error?.message ?? Errors.UNKNOWN
```

Comment:
When using `Optional Chaining` keep in mind the version of typescript that you are handling.
I know we are using an up to date version for this, but this is something to be aware of